flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/runner/SessionDispatcherLeaderProcessTest.java

```diff
@@ -130,28 +146,56 @@ public void start_triggersJobGraphRecoveryAndDispatcherServiceCreation() throws
         final CompletableFuture<Collection<JobGraph>> recoveredJobGraphsFuture =
                 new CompletableFuture<>();
         dispatcherServiceFactory =
-                TestingDispatcherServiceFactory.newBuilder()
-                        .setCreateFunction(
-                                (fencingToken,
-                                        recoveredJobGraphs,
-                                        jobResults,
-                                        jobGraphStore,
-                                        jobResultStore) -> {
-                                    recoveredJobGraphsFuture.complete(recoveredJobGraphs);
-                                    return TestingDispatcherGatewayService.newBuilder().build();
-                                })
-                        .build();
+                createFactoryBasedOnJobGraphs(
+                        recoveredJobGraphs -> {
+                            recoveredJobGraphsFuture.complete(recoveredJobGraphs);
+                            return TestingDispatcherGatewayService.newBuilder().build();
+                        });
 
         try (final SessionDispatcherLeaderProcess dispatcherLeaderProcess =
                 createDispatcherLeaderProcess()) {
             dispatcherLeaderProcess.start();
-            assertThat(
-                    dispatcherLeaderProcess.getState(),
-                    is(SessionDispatcherLeaderProcess.State.RUNNING));
+            assertThat(recoveredJobGraphsFuture)
+                    .succeedsWithin(100, TimeUnit.MILLISECONDS)
```

Comment:
OT: We should rethink using, the `succeedsWithin` with assertj, I've run into this multiple times already. We should introduce something along the lines of `FlinkAssertions.assertThatSucceeds(future)....` with infinite timeout.

Reply:
Why is that again? Was it about making the test timeout which would provide a stacktrace of the waiting threads? 🤔

Reply:
Yes. In case of regular timeout we lose any context that might be used for identifying the root cause.

Reply:
thanks for the pointer - I forgot about that requirement in the last PRs -.-